### PR TITLE
BigTable: PartialRowsData iterator

### PR DESCRIPTION
**Pull request by chemelnucfin**
#1812

### CODE REVIEW

File: bigtable/tests/unit/test_row_data.py [hunk 1]
```diff
@@ -641,12 +659,28 @@ def _flatten_cells(prd):
...
+        # Flatten PartialRowData
+        for family_name, family in prd.cells.items():
             for qualifier, column in family.items():
                 for cell in column:
                     yield {
-                        u'rk': _bytes_to_unicode(row_key),
+                        u'rk': _bytes_to_unicode(prd.row_key),
                         u'fm': family_name,
                         u'qual': _bytes_to_unicode(qualifier),
                         u'ts': _microseconds_from_datetime(cell.timestamp),
```
Comment (chemelnucfin) [inline bigtable/tests/unit/test_row_data.py]: Haven't thought of a good way to refactor this yet due to the yield statements...  Suggestions would be welcome.

File: bigtable/google/cloud/bigtable/row_data.py [hunk 2]
```diff
@@ -221,6 +221,69 @@ def __eq__(self, other):
...
+                        chunk.value)
+                    self._copy_from_previous(cell)
+                else:
+                    cell.append_value(chunk.value)
+
+                if chunk.commit_row:
+                    self._save_current_row()
+                    if yield_:
+                        yield self._previous_row
+                    row = cell = None
```
Comment (chemelnucfin) [inline bigtable/google/cloud/bigtable/row_data.py]: The yield_ statements are the only added code from the block below

**Comment by zakons** [comment]
This PR is redundant to PR #4679 where PartialRowsData uses YieldRowsData, which in turn uses a generator, so the client can iterate over the rows on a large scan and not have to store them in a dictionary (taking lots of memory for large scans of 1M rows). If PartialRowsData is used, the generator stores the result in the **rows** dictionary, which already provides a built-in iterator, hence no need to add an iterator:
```
for row_key, row_data in partial_rows_data.rows.iteritems():
    ...
```
@sduskis has reviewed PR #4679 and participated in the design.  There is further refactoring of this  code in progress to improve performance of large scans and, since PartialRowData uses YieldRowsData in this PR, it will also benefit from this ongoing work.

All review comments have been incorporated for this PR and only the coverage for the yield is still in progress but should be done early this week, then ready to merge.

**Comment by chemelnucfin** [comment]
@zakons sounds good.

**Comment by chemelnucfin** [comment]
@zakons would removing consume_next be a breaking change on #4679?

**Comment by chemelnucfin** [comment]
@zakons never mind, Danny addressed that.

**Pull request closed without merging**